duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/inputscreen/ui/BrowserAndInputScreenTransitionProviderImpl.kt

```diff
@@ -43,49 +44,69 @@ import javax.inject.Inject
+ * is anchored to the same screen coordinates in both the browser screen and input screen.
  */
 @ContributesBinding(scope = AppScope::class)
 class BrowserAndInputScreenTransitionProviderImpl @Inject constructor(
     private val appTheme: AppTheme,
+    private val duckChatInternal: DuckChatInternal,
 ) : BrowserAndInputScreenTransitionProvider {
-
-    override fun getBrowserEnterAnimation(): Int {
-        return if (VERSION.SDK_INT >= 33) {
+    override fun getBrowserEnterAnimation(isTopOmnibar: Boolean): Int =
+        if (VERSION.SDK_INT >= 33) {
             if (appTheme.isLightModeEnabled()) {
-                R.anim.slide_in_from_bottom_fade_in_light
+                if (useTopBar(isTopOmnibar)) {
+                    R.anim.slide_in_from_bottom_fade_in_light
+                } else {
+                    R.anim.fade_in_light
+                }
             } else {
-                R.anim.slide_in_from_bottom_fade_in_dark
+                if (useTopBar(isTopOmnibar)) {
+                    R.anim.slide_in_from_bottom_fade_in_dark
+                } else {
+                    R.anim.fade_in_dark
+                }
             }
         } else {
             R.anim.fade_in
         }
-    }
 
-    override fun getBrowserExitAnimation(): Int {
-        return if (VERSION.SDK_INT >= 33) {
+    override fun getBrowserExitAnimation(isTopOmnibar: Boolean): Int =
+        if (VERSION.SDK_INT >= 33) {
             if (appTheme.isLightModeEnabled()) {
-                R.anim.slide_out_to_bottom_fade_out_light
+                if (useTopBar(isTopOmnibar)) {
+                    R.anim.slide_out_to_bottom_fade_out_light
+                } else {
+                    R.anim.fade_out_light
+                }
             } else {
-                R.anim.slide_out_to_bottom_fade_out_dark
+                if (useTopBar(isTopOmnibar)) {
+                    R.anim.slide_out_to_bottom_fade_out_dark
+                } else {
+                    R.anim.fade_out_dark
+                }
             }
         } else {
             R.anim.fade_out
         }
-    }
 
-    override fun getInputScreenEnterAnimation(): Int {
-        return if (VERSION.SDK_INT >= 33) {
+    override fun getInputScreenEnterAnimation(isTopOmnibar: Boolean): Int =
+        if (VERSION.SDK_INT >= 33 && useTopBar(isTopOmnibar)) {
             R.anim.slide_in_from_top_fade_in
         } else {
             R.anim.fade_in
         }
-    }
 
-    override fun getInputScreenExitAnimation(): Int {
-        return if (VERSION.SDK_INT >= 33) {
+    override fun getInputScreenExitAnimation(isTopOmnibar: Boolean): Int =
+        if (VERSION.SDK_INT >= 33 && useTopBar(isTopOmnibar)) {
             R.anim.slide_out_to_top_fade_out
         } else {
             R.anim.fade_out
         }
-    }
+
+    private fun useTopBar(isTopOmnibar: Boolean): Boolean = isTopOmnibar || !duckChatInternal.inputScreenBottomBarEnabled.value
```

Comment:
Do we need this in two places? It’s also in `InputScreenConfigResolver`

Reply:
Unfortunately, yes. `InputScreenConfigResolver` is tied to `InputScreenActivity` lifecycle because it gets the omnibar position info from an activity param. The transition provider on the other hand also runs in the `BrowserActivity` so the omnibar position is provided independently and `InputScreenConfigResolver` can't be used.

That said, in https://github.com/duckduckgo/Android/pull/6849/commits/ec7f3ea4ce8cddaa96c2e8faba36f4b1f91e303c I added a static function that checks the condition so that if we ever need to update it, it's in a single place.